File: src/NWT_Webshop_Application/Models/Product.cs
﻿using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace NWT_Webshop_Application.Models
{
    public class Product
    {
        [Key]
        public int ProductID { get; set; }

        [Required]
        [StringLength(30, MinimumLength = 5)]
        public string Name { get; set; }

        [Required]
        [StringLength(500)]
        [DataType(DataType.Text)]
        public string Description { get; set; }

        [Required]
        [StringLength(200)]
        [DataType(DataType.Url)]
        public string ImageURL { get; set; }

        [Required]
        public float Price { get; set; }

        [Display(Name = "Average Score")]
        public float AverageScore { get; set; }

        [StringLength(500)]
        public string Scores { get; set; }

        [Required]
        [StringLength(200)]
        public string Tags { get; set; }

        [JsonIgnore]
        public virtual ICollection<ShoppingCartProduct> ShoppingCartProducts { get; set; }

    }
}

File: src/NWT_Webshop_Application/App_Client/src/app/models/Product.ts
export default class Product {
    public id: number;
    public name: string;
    public description: string;
    public imageURL: string;
    public price: number;
    public averageScore: number;
    public scores: string[];
    public amount: number;
    public tags: string[];

    constructor(prodObj: ProductConstrucorObject){
        this.id = prodObj.id;
        this.name = prodObj.name;
        this.description = prodObj.description;
        this.imageURL = prodObj.imgURL;
        this.price = prodObj.price;
        this.averageScore = prodObj.avgScore;
        this.scores = prodObj.scores;
        this.amount = prodObj.amount;
        this.tags = prodObj.tags;
    }
}


interface ProductConstrucorObject {
    id: number;
    name: string;
    description: string;
    imgURL: string;
    price: number;
    avgScore: number;
    scores: string[]
    amount: number;
    tags: string[];
}